### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by NNKamel**

### CODE REVIEW

File: AdminToolsTests/CustomApplicationFactory.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                    OptionsBuilderConfigurationExtensions.TokenValidationParameters = new TokenValidationParameters()
+                    {
+                        IssuerSigningKey = FakeJwtManager.SecurityKey,
+                        ValidIssuer = FakeJwtManager.Issuer,
+                        ValidAudience = FakeJwtManager.Audience
+                    };
+                });
+            });
+        }
+    }
```
Comment (clarson217) [inline AdminToolsTests/CustomApplicationFactory.cs:33]: Please add a comment on what this is.

**Review by clarson217 — CHANGES_REQUESTED** [review]

File: AdminToolsTests/CustomApplicationFactory.cs (hunk 1 above)
Comment (NNKamel) [reply AdminToolsTests/CustomApplicationFactory.cs:33]: Done

**Review by clarson217 — APPROVED** [review]

File: AdminToolsLogic/Logic/TopicLogic.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        private readonly TopicsRepository _repo;
+        private readonly Mapper _mapper;
+
+        public TopicLogic(TopicsRepository _repo, Mapper _mapper)
+        {
+            this._repo = _repo;
+            this._mapper = _mapper;
+        }
+        public List<AffectedSubject> GetSubjects()
```
Comment (markmoorerev) [inline AdminToolsLogic/Logic/TopicLogic.cs:20]: comments

File: AdminToolsModels/LogicModels/ReportModel.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+using System;
+
+namespace AdminToolsModels.LogicModels
+{
+    public class ReportModel
```
Comment (markmoorerev) [inline AdminToolsModels/LogicModels/ReportModel.cs:5]: purpose?

File: AdminToolsTests/FakeJwtManager.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
+//source: https://auth0.com/blog/xunit-to-test-csharp-code/
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using Microsoft.IdentityModel.Tokens;
+
+namespace AdminToolsTests
+{
+    public static class FakeJwtManager
```
Comment (markmoorerev) [inline AdminToolsTests/FakeJwtManager.cs:10]: purpose?

File: Repository/Models/AffectedSubject.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+
+#nullable disable
+
+namespace Repository.Models
+{
+    public partial class AffectedSubject
```
Comment (markmoorerev) [inline Repository/Models/AffectedSubject.cs:8]: purpose?

File: Repository/Models/Cinephiliacs_AdmintoolsContext.cs [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        public virtual DbSet<AffectedSubject> AffectedSubjects { get; set; }
+        public virtual DbSet<ResolvedTicket> ResolvedTickets { get; set; }
+        public virtual DbSet<Ticket> Tickets { get; set; }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            if (!optionsBuilder.IsConfigured)
+            {
+#warning To protect potentially sensitive information in your connection string, you should move it out of source code. You can avoid scaffolding the connection string by using the Name= syntax to read it from configuration - see https://go.microsoft.com/fwlink/?linkid=2131148. For more guidance on storing connection strings, see http://go.microsoft.com/fwlink/?LinkId=723263.
+                optionsBuilder.UseSqlServer("Server=tcp:cinephiliacs.database.windows.net,1433;Initial Catalog=Cinephiliacs_Admintools;Persist Security Info=False;User ID=kugelsicher;Password=F36UWevqvcDxEmt;MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;");
```
Comment (markmoorerev) [inline Repository/Models/Cinephiliacs_AdmintoolsContext.cs:29]: this should be in usersecrets for dev and configured on Azure for prod.

File: Repository/Models/ResolvedTicket.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+
+#nullable disable
+
+namespace Repository.Models
+{
+    public partial class ResolvedTicket
```
Comment (markmoorerev) [inline Repository/Models/ResolvedTicket.cs:8]: purpose?

File: Repository/Models/Ticket.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+
+#nullable disable
+
+namespace Repository.Models
+{
+    public partial class Ticket
```
Comment (markmoorerev) [inline Repository/Models/Ticket.cs:8]: purpose?

File: Repository/TopicsRepository.cs [hunk 9]
```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AdminToolsModels.LogicModels;
+using Repository.Models;
+
+namespace Repository
+{
+    public class TopicsRepository
```
Comment (markmoorerev) [inline Repository/TopicsRepository.cs:10]: purpose?

**Review by markmoorerev — CHANGES_REQUESTED** [review]
comments and connection string

**Pull request closed without merging**